### PR TITLE
LYN-7637 Allow Terrain Physics Collider to assign a default physics material

### PR DESCRIPTION
**Pull request by SergeyAMZN**
Signed-off-by: Sergey Pereslavtsev <pereslav@amazon.com>

### CODE REVIEW

File: Gems/Terrain/Code/Source/Components/TerrainPhysicsColliderComponent.cpp [hunk 1]
```diff
@@ -82,6 +83,8 @@ namespace Terrain
                     ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
                     ->Attribute(AZ::Edit::Attributes::Visibility, AZ::Edit::PropertyVisibility::ShowChildrenOnly)
                     ->Attribute(AZ::Edit::Attributes::AutoExpand, true)
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &TerrainPhysicsColliderConfig::m_defaultMaterialSelection,
+                        "Default Surface Physics Material", "Select a material to be used by maps surfaces by default")
```
Comment (mbalfour-amzn) [inline Gems/Terrain/Code/Source/Components/TerrainPhysicsColliderComponent.cpp:87]: "to be used by maps surfaces by default" -> 
maybe either "to be used as the default surface material" or "to be used by unmapped surfaces by default"?

File: Gems/Terrain/Code/Tests/TerrainPhysicsColliderTests.cpp [hunk 2]
```diff
@@ -502,3 +502,95 @@ TEST_F(TerrainPhysicsColliderComponentTest, TerrainPhysicsColliderGetHeightsAndM
...
+    const Physics::MaterialId mat1 = Physics::MaterialId::Create();
+
+    const SurfaceData::SurfaceTag tag1 = SurfaceData::SurfaceTag("tag1");
+    const SurfaceData::SurfaceTag tag2 = SurfaceData::SurfaceTag("tag2");
+
+    Terrain::TerrainPhysicsSurfaceMaterialMapping mapping1;
+    mapping1.m_materialId = mat1;
+    mapping1.m_surfaceTag = tag1;
+    config.m_surfaceMaterialMappings.emplace_back(mapping1);
+    config.m_defaultMaterialSelection.SetMaterialId(defaultSurfaceMaterial);
```
Comment (mbalfour-amzn) [inline Gems/Terrain/Code/Tests/TerrainPhysicsColliderTests.cpp:526]: Suggestion: Might want to make a second similar test where there aren't _any_ mappings set up, to verify there aren't any hidden dependencies where at least one mapped material needs to exist.

**Review by mbalfour-amzn — APPROVED** [review]

**Review by cgalvan — APPROVED** [review]

File: Gems/Terrain/Code/Source/Components/TerrainPhysicsColliderComponent.cpp (hunk 1 above)
Comment (moraaar) [reply Gems/Terrain/Code/Source/Components/TerrainPhysicsColliderComponent.cpp:87]: I like @mbalfour-amzn suggestion "to be used by unmapped surfaces by default", it explicitly says when the default material will be used.

File: Gems/Terrain/Code/Tests/TerrainPhysicsColliderTests.cpp (hunk 2 above)
Comment (moraaar) [reply Gems/Terrain/Code/Tests/TerrainPhysicsColliderTests.cpp:526]: Suggestion: It'd be good to have another test where default surface material in `TerrainPhysicsColliderConfig` is not set, to check nothing breaks if the user doesn't set a default material to the component.

**Review by moraaar — APPROVED** [review]

File: Gems/Terrain/Code/Tests/TerrainPhysicsColliderTests.cpp (hunk 2 above)
Comment (SergeyAMZN) [reply Gems/Terrain/Code/Tests/TerrainPhysicsColliderTests.cpp:526]: @moraaar This is already tested in the cases above (i.e. the existing tests). 
@mbalfour-amzn Yep, good idea! Adding that

File: Gems/Terrain/Code/Source/Components/TerrainPhysicsColliderComponent.cpp (hunk 1 above)
Comment (SergeyAMZN) [reply Gems/Terrain/Code/Source/Components/TerrainPhysicsColliderComponent.cpp:87]: Nice, updated the wording, thanks!

**Review by moraaar — APPROVED** [review]

**Pull request merged**